src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +238,92 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+char* SystemNative_GetGroupName(uint32_t gid)
+{
+    struct group* result;
+    char* name;
```

Comment:
(nit) I usually prefer to declare variables close to where they are used. I think it helps make the code readable. If this were me, I would have just duplicated the declarations in the different `#if` cases.